litmus-portal/authentication/api/routes/user_router.go

```diff
@@ -0,0 +1,184 @@
+package routes
+
+import (
+	"litmus/litmus-portal/authentication/api/middleware"
+	"litmus/litmus-portal/authentication/api/presenter"
+	"litmus/litmus-portal/authentication/pkg/entities"
+	"litmus/litmus-portal/authentication/pkg/user"
+	"litmus/litmus-portal/authentication/pkg/utils"
+
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+//UserRouter creates all the required routes for user authentications purposes.
+func UserRouter(router *gin.Engine, service user.Service) {
+	router.POST("/login", loginUser(service))
+	router.Use(middleware.JwtMiddleware())
+	router.POST("/update/password", updatePassword(service))
+	router.POST("/reset/password", resetPassword(service))
+	router.POST("/create", createUser(service))
+	router.POST("/update/details", updateUser(service))
```

Comment:
Changing the protocol will required changes from the frontend and as mentioned above, the server was re-written keeping in mind that there should be no changes in request/response as in the current authentication-server used by litmus-frontend. I would prefer PUT over POST. Had to choose PUT to avoid further changes in litmus-frontend.

Reply:
@DarthBenro008 its better to change this cc: @rajdas98 @gdsoumya

Reply:
@ishangupta-ds , I would like to confirm,  both the `/update/details` and `/update/password` routes would have `PUT` requests if we will be going ahead with it? 

## New Routes Table

| Type of Request 	|      Routes      	|               Functionality              	| Authorisation Required 	|
|-----------------	|:----------------:	|:----------------------------------------:	|------------------------	|
| POST            	| /login           	| Logs in the user and generates JWT Token 	| No                     	|
| GET             	| /users           	| Fetches all the users to be displayed    	| Yes                    	|
| POST            	| /create          	| Allows admin to create a user            	| Yes (with admin role)  	|
| PUT            	| /update/password 	| Update password of logged in user        	| Yes                    	|
| POST            	| /reset/password  	| Allows admins to reset user's password   	| Yes (with admin role)  	|
| PUT            	| /update/details  	| Allows user to update email and name     	| Yes                    	|

Reply:
Ideally both should be updated @DarthBenro008 , awaiting comments from @rajdas98 and @gdsoumya too.

Reply:
IMO, If we use query params in `/update/details` and  `/update/password`, then having `PUT` method makes sense. Otherwise, It will not make any difference if we use  `POST`.

Reply:
Let's leave it as a POST request for now, we can update it later along with the frontend.